localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/mock.py

```diff
@@ -66,45 +66,37 @@ def parse_invalid_json(self, body: str) -> dict:
         try:
             statuscode = ""
             matched = re.match(r"^\s*{(.+)}\s*$", body).group(1)
-            splits = [m.strip() for m in matched.split(",")]
+            pairs = [m.strip() for m in matched.split(",")]
             # TODO this is not right, but nested object would otherwise break the parsing
-            kvs = [s.split(":", maxsplit=1) for s in splits]
-            for kv in kvs:
-                assert len(kv) == 2
-                k, v = kv
-                k = k.strip()
-                v = v.strip()
-
-                assert k
-                assert v
-
-                if k == "statusCode":
-                    statuscode = int(v)
+            key_values = [s.split(":", maxsplit=1) for s in pairs]
+            for key_value in key_values:
+                assert len(key_value) == 2
+                key, value = [el.strip() for el in key_value]
+                key = key or '""'
+                value = value or '""'
+
+                assert (leading_key_char := key[0]) not in "[{"
+                if leading_key_char in "'\"":
+                    assert len(key) >= 2
+                    assert key[-1] == leading_key_char
+
+                if key == "statusCode":
+                    statuscode = int(value)
```

Comment:
I think we had the mistake in our current implementation too, but here we directly check against the `stateCode` value without regards to single or double quotes it seems (except of the leftover second `if` check)

I think we need to still address that 🤔

Reply:
oh yes! I must have been tired when I wrote made the cleanup! The goal of moving it under the quotes assertions, was to solve this very issue! But it isn't removing the quotes anymore as we originally had :facepalm: